code/modules/mob/abstract/freelook/ai/eye.dm

```diff
@@ -97,3 +97,33 @@
 
 	eyeobj.acceleration = !eyeobj.acceleration
 	to_chat(usr, "Camera acceleration has been toggled [eyeobj.acceleration ? "on" : "off"].")
+
+
+/mob/living/silicon/ai/proc/play_sound()
+	set category = "AI Commands"
+	set name = "Play Music In The Area"
+
+	var/delay = world.time - time_music
+	if(delay < 3000)
+		delay \= 10
+		to_chat(src, "<span class='warning'>You have played music recently, you have to wait [(round(delay/60) > 0) ? ("[round(delay/60)] minutes") : ("[delay] seconds") ]</span>")
+		return
+	var/list/sounds = file2list("sound/serversound_list.txt");
+	sounds += "--CANCEL--"
+	sounds += sounds_cache
+
+	var/melody = input("Select a sound from the server to play", "Server sound list", "--CANCEL--") in sounds

```

Comment:
Could just go with `as text|null in sounds`, which provides a cancel button. And then check if `if (!melody) return`